pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/model/config/ChatCompletionsConfig.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.pulsar.functions.transforms.model.config;
+
+import com.azure.ai.openai.models.ChatMessage;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import java.util.Map;
+import lombok.Data;
+
+@Data
+public class ChatCompletionsConfig extends StepConfig {
```

Comment:
Can the user select a certain field on the topic as input to the completion step?

Reply:
Yes. Mustache templating can be used in the chat messages.
The model used for templating is
```java
@Data
public class JsonRecord {
  String topicName;
  String destinationTopic;
  Object key;
  Object value;
  Map<String, String> properties;
  Long eventTime;
}
``` 
so you can do `The topic is {{ destinationTopic }}.`
If value/key is AVRO in the record, they are converted to `Map<String, Object>` in the JsonRecord.